airbyte-integrations/connectors/source-relational-db/src/main/java/io/airbyte/integrations/source/relationaldb/state/GlobalStateManager.java

```diff
@@ -99,6 +102,26 @@ private CdcState extractCdcState(final AirbyteStateMessage airbyteStateMessage)
     }
   }
 
+  private Set<AirbyteStreamNameNamespacePair> extractStreams(final AirbyteStateMessage airbyteStateMessage) {
+    if (airbyteStateMessage.getType() == AirbyteStateType.GLOBAL) {
+      return airbyteStateMessage.getGlobal().getStreamStates().stream()
+          .map(streamState -> {
+            final AirbyteStreamState cloned = Jsons.clone(streamState);
+            return new AirbyteStreamNameNamespacePair(cloned.getStreamDescriptor().getName(), cloned.getStreamDescriptor().getNamespace());
+          }).collect(Collectors.toSet());
+    } else {
```

Comment:
checking my understanding: in this case, we expect that type is null, because we should never be emitting per-stream state?

Reply:
Yes correct. It can either be global or null which denotes legacy